src/mpid/ch4/generic/am/mpidig_am_send.h

```diff
@@ -79,8 +85,8 @@ static inline int MPIDIG_isend_impl(const void *buf, MPI_Aint count, MPI_Datatyp
         return MPIDIG_do_ssend(buf, count, datatype, rank, tag, comm, context_offset,
                                addr, request, errflag);
     } else if (data_sz > mpidig_eager_limit(is_local)) {
-        return MPIDIG_do_rndv_send(buf, count, datatype, data_sz, rank, tag, comm, context_offset,
-                                   addr, request, errflag);
+        return MPIDIG_do_lmt_send(buf, count, datatype, data_sz, rank, tag, comm, context_offset,
+                                  addr, request, errflag);
```

Comment:
Isn't it simpler to test the `mem_type` here and dispatch to either `rndv_send` or `gpu_send` ? I think it is easier to read if we keep all dispatches in a single place.